### PR TITLE
EXSWHTEC-281 - Implement Unit Tests for assert functions

### PR DESCRIPTION
**Pull request by milos-mozetic**
Positive and negative tests are implemented for:
- assert, i.e. run-time assert
- static_assert, i.e. compile-time assert

### CODE REVIEW

File: catch/unit/assertion/assert.cc [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+  assert(tid % 2 == 1);
+}
+
+template <bool should_abort> void LaunchAssertKernel() {
+  const int num_blocks = 2;
+  const int num_threads = 16;
+
+  if constexpr (should_abort) {
+    AssertFailKernel<<<num_blocks, num_threads, 0, 0>>>();
+#if HT_AMD
```
Comment (chrispaquot) [inline catch/unit/assertion/assert.cc:67]: add the test to json to skip it on AMD.

Comment (MMozetic) [reply catch/unit/assertion/assert.cc:67]: @chrispaquot: Can you please describe what is the issue here? Additionally, is it causing issues on Windows or Linux?

Comment (milos-mozetic) [reply catch/unit/assertion/assert.cc:67]: @chrispaquot: Can you please describe what is the issue here? Additionally, is it problematic on Windows or Linux?

**Comment by milos-mozetic** [comment]
@rakesroy, please note that the issue with this PR is resolved within commit: https://github.com/ROCm-Developer-Tools/hip-tests/pull/210/commits/557446b622e447dcf1e6e4b278d959fcc2bef7d1. Also, some files where not formatted with clang-format so that can be observed also in the diff.
Can you please check again that the problem is solved now?

**Comment by rakesroy** [comment]
@milos-mozetic ,
Due to recent pushes to develop branch, merge conflicts occurred in this PR.
Will you please resolve them?

**Comment by milos-mozetic** [comment]
@rakesroy: Merge conflict has been resolved within commit https://github.com/ROCm-Developer-Tools/hip-tests/pull/210/commits/73fdae115f219231ea88e28fdee30812f455a929.

**Comment by rakesroy** [comment]
PR has been merged into develop branch via commit 044a59496cbfec069485d9233d93d96f46927565.

**Pull request closed without merging**